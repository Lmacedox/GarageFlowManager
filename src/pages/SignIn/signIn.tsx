import { Typography } from '@/components/typography';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';

import { InputErrorMessage } from '@/components/input-error-message';
import { useSignIn } from './use-signIn';
import { LoaderCircle } from 'lucide-react';

export function SignIn() {
  const { onSubmit, register, errors, isPending, sucess } = useSignIn();

  return (
    <div className="w-full text-center">
      <Typography.H2>Acessar Painel</Typography.H2>
      <Typography.P className="text-muted-foreground">
        Acompanhe o fluxo da sua garagem!
      </Typography.P>

      <form onSubmit={onSubmit} className="mt-11 flex flex-col gap-2">
        <div className="space-y-5 text-left">
          <Input placeholder="Nome do cliente" {...register('userName')} />
          <InputErrorMessage inputName={'userName'} errors={errors} />
        </div>

        <div className="space-y-5 text-left">
          <Input
            placeholder="Senha"
            type="password"
            {...register('password')}
          />
          <InputErrorMessage inputName={'password'} errors={errors} />
        </div>

        <Button className="mt-1 w-full" disabled={isPending}>
          {isPending ? <LoaderCircle className="animate-spin" /> : 'Entrar'}
        </Button>

        <h1>{sucess}</h1>
      </form>
    </div>
  );
}
