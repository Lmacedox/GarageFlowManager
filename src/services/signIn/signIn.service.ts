import { signInForm } from '@/app/signin/signIn.type'

export class SignInHttpRequests {
  async SignInService(userData: signInForm) {
    console.log('@UserData', userData)
  }
}
