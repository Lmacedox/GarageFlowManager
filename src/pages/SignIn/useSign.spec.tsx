import { queryClient } from '@/lib/react-query';
import { QueryClientProvider } from '@tanstack/react-query';
import { act, fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import userEvent from '@testing-library/user-event';

import { SignIn } from './signIn';

interface IWrapper {
  children: React.ReactNode;
}

describe('SignIn', () => {
  const wrapper = render(<SignIn />, {
    wrapper: ({ children }: IWrapper) => {
      return (
        <MemoryRouter>
          <QueryClientProvider client={queryClient}>
            {children}
          </QueryClientProvider>
        </MemoryRouter>
      );
    },
  });

  // it('Should has feedback to required fields to send empty form fields', async () => {
  //   const submitButton = wrapper.getByRole('button');

  //   await act(async () => {
  //     fireEvent.click(submitButton);
  //   });

  //   const textElement = wrapper.queryAllByText('Campo obrigatório');

  //   expect(textElement.length).equals(2);
  // });

  it('Should sucess in authentication', async () => {
    const user = userEvent.setup();
    const userName = wrapper.getByPlaceholderText(
      /Nome do cliente/i,
    ) as HTMLInputElement;

    const password = wrapper.getByPlaceholderText(/Senha/i) as HTMLInputElement;

    fireEvent.change(userName, { target: { value: 'teste' } });
    fireEvent.change(password, { target: { value: 'pass1' } });

    const submitButton = wrapper.getByRole('button');

    // await act(async () => {
    //   fireEvent.click(submitButton);
    // });

    await user.click(submitButton);

    // Aguarda a navegação para a nova rota
    expect(wrapper.asFragment()).toMatchSnapshot();
    // await wrapper.findByText(/DashBoard/i);

    // expect(wrapper.getByText(/DashBoard/i)).toBeInTheDocument();
  });
});
