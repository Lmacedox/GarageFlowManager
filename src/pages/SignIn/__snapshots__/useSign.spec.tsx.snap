// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignIn > Should sucess in authentication 1`] = `
<DocumentFragment>
  <div
    class="w-full text-center"
  >
    <h2
      class="scroll-m-20 pb-2 text-3xl font-semibold tracking-tight first:mt-0"
    >
      Acessar Painel
    </h2>
    <p
      class="leading-7 text-muted-foreground"
    >
      Acompanhe o fluxo da sua garagem!
    </p>
    <form
      class="mt-11 flex flex-col gap-2"
    >
      <div
        class="space-y-5 text-left"
      >
        <input
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          name="userName"
          placeholder="Nome do cliente"
          value=""
        />
      </div>
      <div
        class="space-y-5 text-left"
      >
        <input
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          name="password"
          placeholder="Senha"
          type="password"
          value=""
        />
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mt-1 w-full"
        disabled=""
      >
        <svg
          class="lucide lucide-loader-circle animate-spin"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12a9 9 0 1 1-6.219-8.56"
          />
        </svg>
      </button>
      <h1 />
    </form>
  </div>
</DocumentFragment>
`;
